### PR TITLE
Migrate `hausaufgabenheft_logik` and `firebase_hausaufgabenheft_logik` to null safety.

### LINKED ISSUE #191: Migrate the `hausaufgabenheft_logik` package to null safety
Blocked by:
* #203 
* #206

### LINKED ISSUE #194: Migrate the `firebase_hausaufgaben_logik` package to null safety
Blocked by:
* #187 
* #191 
* #203 
* #184

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Requires #743 
Requires #738 

Closes #191 
Closes #194

### CODE REVIEW

**Comment by Jonas-Sander** [comment]
@Jonas-Sander The changes of the required PRs are already inside this PR, so you should probably wait until these PRs are merged so that the diff of this PR is only about the changes I made here.

**Comment by Jonas-Sander** [comment]
Hm in the web preview the groups are not loaded correctly. Don't know if this comes from this PR or some of ones before.

**Review by nilsreichardt — APPROVED** [review]
LGTM

**Pull request merged**
